src/js/actions/layers.js

```diff
@@ -451,9 +451,10 @@ define(function (require, exports) {
      *  non-lazy properties updated, and will be returned to an uninitialized
      *  state. This is a performance optimization. NOTE: the selection state
      *  of the layer models passed in must be accurate for this to work correctly!
+     * @param {boolean} skipResetBorderPolicy If truthy do not reset Border Policies
      * @return {Promise}
      */
-    var resetLayers = function (document, layers, amendHistory, lazy) {
+    var resetLayers = function (document, layers, amendHistory, lazy, skipResetBorderPolicy) {
```

Comment:
As mentioned offline, it seems like it might be better to remove this flag and just have `tools.resetBorderPolicies` be a no-op in mask mode. That seems preferable to this, but if that doesn't work then this is OK.